src/main/java/com/thoughtworks/aceleradora/validadores/ListaValidador.java

```diff
@@ -0,0 +1,63 @@
+package com.thoughtworks.aceleradora.validadores;
+
+import com.thoughtworks.aceleradora.dominio.MinhaLista;
+import com.thoughtworks.aceleradora.repositorios.MinhaListaRepositorio;
+import com.thoughtworks.aceleradora.validadores.anotacoes.ListaValida;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Optional;
+
+public class ListaValidador implements ConstraintValidator<ListaValida, MinhaLista> {
+    @Autowired
+    MinhaListaRepositorio repositorio;
+
+    @Override
+    public void initialize(ListaValida constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(MinhaLista minhaLista, ConstraintValidatorContext context) {
+        context.disableDefaultConstraintViolation();
+
+        return nomeAindaNaoExisteNoBanco(minhaLista, context)
+                && nomeNaoEstaVazio(minhaLista, context)
+                && listaContemProdutos(minhaLista, context);
+    }
+
+    private boolean nomeAindaNaoExisteNoBanco(MinhaLista lista, ConstraintValidatorContext context) {
+        Optional<MinhaLista> listaComConflito = repositorio.findByNome(lista.getNome());
+
+        if(listaComConflito.isPresent() && !listaComConflito.get().getId().equals(lista.getId())) {
```

Comment:
Não daria pra usar o `equals` aqui?

```java
boolean ehMesmaLista = listaComConflito.map(l -> l.equals(lista)).orElse(false);

if(ehMesmaLista) {
  // ...
}
```

To extrapolando com um `map`, mas a ideia em volta do `equals` mesmo hehehe.

Reply:
É que a lista com update pode não ter o mesmos produtos, significando que elas vão ser diferentes quando ainda são a mesma lista. Podemos modificar o equals pra não levar isso em conta.

Reply:
Vou fazer isso em outro PR pq o pessoal precisa disso pro showcase.